libarchive/archive_read_support_format_rar.c

```diff
@@ -2176,6 +2176,19 @@ read_data_compressed(struct archive_read *a, const void **buff, size_t *size,
     {
       start = rar->offset;
       end = start + rar->dictionary_size;
+
+      /* We don't want to overflow the window and overwrite data that we write
+       * at 'start'. Therefore, reduce the end length by the maximum match size,
+       * which is 260 bytes. You can compute this maximum by looking at the
+       * definition of 'expand', in particular when 'symbol >= 271'. */
+      /* NOTE: It's possible for 'dictionary_size' to be less than this 260
+       * value, however that will only be the case when 'unp_size' is small,
+       * which should only happen when the entry size is small and there's no
+       * risk of overflowing the buffer */
+      if (rar->dictionary_size > 260) {
```

Comment:
One thing for consideration: this could be negatively impactful for small file sizes. E.g. if a file is 511 bytes, `dictionary_size` will be 512 (smallest power of two greater than 260), meaning this would take at least two passes. Two passes isn't that much, so maybe trying to do something more here is over-optimizing a bit too much.